src/Input/Mouse.php

```diff
@@ -260,26 +262,54 @@ private function scrollToBoundary(int $right, int $bottom): self
      * @return $this
      */
     public function find(string $selectors, int $position = 1): self
+    {
+        $this->findElement(new CssSelector($selectors), $position);
+
+        return $this;
+    }
+
+    /**
+     * Find an element and move the mouse to a random position over it.
+     *
+     * The search could result in several elements. The $position param can be used to select a specific element.
+     * The given position can only be between 1 and the maximum number or elements. It will be adjusted to the
+     * minimum and maximum values if needed.
+     *
+     * Example:
+     * $page->mouse()->findElement(new CssSelector('#a')):
+     * $page->mouse()->findElement(new CssSelector('.a'), 2);
+     * $page->mouse()->findElement(new XPathSelector('//*[@id="a"]'), 2);
+     *
+     * @param Selector $selector selector to use
+     * @param int      $position (optional) which element of the result set should be used
+     *
+     * @throws \HeadlessChromium\Exception\CommunicationException
+     * @throws \HeadlessChromium\Exception\NoResponseAvailable
+     * @throws \HeadlessChromium\Exception\ElementNotFoundException
+     *
+     * @return $this
+     */
+    public function findElement(Selector $selector, int $position = 1): self
     {
         $this->page->assertNotClosed();
 
         try {
-            $elementList = $this->page
-                ->evaluate('JSON.parse(JSON.stringify(document.querySelectorAll("'.$selectors.'")));')
+            $elementCount = $this->page
+                ->evaluate(\sprintf('JSON.parse(JSON.stringify(%s));', $selector->expressionCount()))
                 ->getReturnValue();
 
-            $position = \max(0, ($position - 1));
-            $position = \min($position, (\count($elementList) - 1));
+            $position = \max(1, $position);
+            $position = \min($position, $elementCount);
 
             $element = $this->page
-                ->evaluate('JSON.parse(JSON.stringify(document.querySelectorAll("'.$selectors.'")['.$position.'].getBoundingClientRect()));')
+                ->evaluate(\sprintf('JSON.parse(JSON.stringify(%s.getBoundingClientRect()));', $selector->expressionFindOne($position)))
                 ->getReturnValue();
         } catch (JavascriptException $exception) {
-            throw new ElementNotFoundException('The search for "'.$selectors.'" returned no elements.');
+            throw new ElementNotFoundException('The search for "'.$selector->expressionCount().'" returned no result.');
```

Comment:
The message for this exception will not mention the selector name, will it?

Reply:
https://github.com/chrome-php/chrome/pull/340#discussion_r795632223